rasa_core/policies/augmented_memoization.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+import typing
+
+from typing import Optional, Any, Dict, List, Text
+
+from rasa_core.policies.memoization import MemoizationPolicy
+from rasa_core.events import ActionExecuted
+
+logger = logging.getLogger(__name__)
+
+if typing.TYPE_CHECKING:
+    from rasa_core.trackers import DialogueStateTracker
+    from rasa_core.domain import Domain
+
+ENABLE_FEATURE_STRING_COMPRESSION = True
+
+
+class AugmentedMemoizationPolicy(MemoizationPolicy):
```

Comment:
yes, because Memoization sets the probability to `1`, the fact that it does additional staff might create confusion, so I split them. This AugmentedMemoizationPolicy is more elaborate version of scoring policy